transportation_splitter.py

```diff
@@ -7,34 +7,73 @@
 DESTINATIONS_COLUMN = "destinations"
+LR_SCOPE_KEY = "between"
+"""
+IS_ON_SUB_SEGMENT_THRESHOLD_METERS 1mm
+This is the max distance from a connector point to a sub-segment for the connector to be 
+considered "on" that sub-segment. It is used for the fallback logic to find connector's 
+place in the segment's coordinates, for dealing with edge cases where the exact connector 
+latlong coordinates are not found on the geometry because of rounding.
+"""
+IS_ON_SUB_SEGMENT_THRESHOLD_METERS = 0.001
+
+class SplitConfig(NamedTuple):
+    """
+    Controls wether or not to introduce a split on every connector along the segment
+    """
+    split_at_connectors: bool = True
 
-# List of columns that are considered for identifying the LRs values to split at is constructed at runtime out of input parquet's schema columns that have LR_SCOPE_KEY = "between" anywhere in their structure. The include/exclude constants below are applied to that list.
-# Explicit list of column names to use for finding LR "between" values to split at. Set to None or empty list to use all columns that have "between" fields.
-LR_COLUMNS_TO_INCLUDE = []
-# list columns to exclude from finding the LR values to split at. Set to None or empty list to use all columns that have "between" fields.
-LR_COLUMNS_TO_EXCLUDE = [] 
+    """
+    Which columns to explicitly include when looking for 'between' LR values to split at.
+    If left emtpy all then columns in the input parquet are considered.
+    If non-empty list is provided, then only those columns are considered.
+    """
+    lr_columns_to_include: list[str] = []
 
-LR_SCOPE_KEY = "between"
-POINT_PRECISION = 7
-LR_SPLIT_POINT_MIN_DIST_METERS = 0.01 # New split points are needed for linear references. How far in meters from other existing splits (from either connectors or other LRs) do these LRs need to be for us to create a new connector for them instead of using the existing connector.
-IS_ON_SUB_SEGMENT_THRESHOLD_METERS = 0.001 # 1mm -> max distance from a connector point to a sub-segment for the connector to be considered "on" that sub-segment
+    """
+    Which columns to explicitly exclude when looking for 'between' LR values to split at.
+    Behaves like the include counterpart but negative.
+    """
+    lr_columns_to_exclude: list[str] = []
+
+    """
+    How many digits to round the lat and long for split points.
+    """
+    point_precision: int = 7
+
+    """
+    New split points are needed for linear references. This controls how far in meters from 
+    other existing splits (from either connectors or other LRs) do these LRs need to be 
+    for us to create a new connector for them instead of using the existing connector.
+    """
+    lr_split_point_min_dist_meters: float = 0.01 # 1cm
+
+    """
+    Skips steps for which intermediate streams are found, default True, set to False to always force reprocess all sub-steps 
+    """
+    reuse_existing_intermediate_outputs: bool = True
```

Comment:
Do you think it's feasible to run this without materializing intermediate steps? It would run a lot faster overall